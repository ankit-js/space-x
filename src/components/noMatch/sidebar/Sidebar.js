import React from "react";
import { yearFilterData } from "../../../utils/meta"

const SideBarComponent = ({
  _handleFilters,
  activeYear,
  land_success,
  launch_success,
}) => (
  <aside className="sidebar">
    <h2>Filters</h2>
    <div className="sidebarContent">
      <div className="sidebarTitle">Launch Year</div>
      <ul className="sidebarList">
        {
          yearFilterData.map((yearFilter) => {
            return (
              <li key={yearFilter.id}>
                {
                  yearFilter.child.map(data => {
                    return (
                      <span
                        key={data.id}
                        onClick={() => _handleFilters("activeYear", data.value)}
                        className={activeYear === data.value ? "active" : ""}
                      >
                        {data.title}
                      </span>
                    )
                  })
                }
              </li>
            )
          })
        }
      </ul>
      <div className="sidebarTitle">Successful Launch</div>
      <ul className="sidebarList">
        <li>
          <span
            onClick={() => _handleFilters("launch_success", true)}
            className={launch_success === true ? "active" : ""}
          >
            True
          </span>
          <span
            onClick={() => _handleFilters("launch_success", false)}
            className={launch_success === false ? "active" : ""}
          >
            False
          </span>
        </li>
      </ul>
      <div className="sidebarTitle">Successful Landing</div>
      <ul className="sidebarList">
        <li>
          <span
            onClick={() => _handleFilters("land_success", true)}
            className={land_success === true ? "active" : ""}
          >
            True
          </span>
          <span
            onClick={() => _handleFilters("land_success", null)}
            className={land_success === null ? "active" : ""}
          >
            False
          </span>
        </li>
      </ul>
    </div>
  </aside>
);

export default SideBarComponent;
