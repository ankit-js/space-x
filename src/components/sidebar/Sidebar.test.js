import React from 'react';
import { shallow } from 'enzyme';
import Sidebar from './Sidebar';

let wrapped = shallow(<Sidebar></Sidebar>);
let title = "Filters"

describe('Side bar component', () => {
  it('should render the Title Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('renders the Title', () => { 
    expect(wrapped.find('h2').text()).toEqual(title);
  });
});