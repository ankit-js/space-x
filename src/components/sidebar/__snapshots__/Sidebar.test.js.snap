// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side bar component should render the Title Component correctly 1`] = `
<aside
  className="sidebar"
>
  <h2>
    Filters
  </h2>
  <div
    className="sidebarContent"
  >
    <div
      className="sidebarTitle"
    >
      Launch Year
    </div>
    <ul
      className="sidebarList"
    >
      <li
        key="2006-2007"
      >
        <span
          className=""
          key="2006"
          onClick={[Function]}
        >
          2006
        </span>
        <span
          className=""
          key="2007"
          onClick={[Function]}
        >
          2007
        </span>
      </li>
      <li
        key="2008-2009"
      >
        <span
          className=""
          key="2008"
          onClick={[Function]}
        >
          2008
        </span>
        <span
          className=""
          key="2009"
          onClick={[Function]}
        >
          2009
        </span>
      </li>
      <li
        key="2010-2011"
      >
        <span
          className=""
          key="2010"
          onClick={[Function]}
        >
          2010
        </span>
        <span
          className=""
          key="2011"
          onClick={[Function]}
        >
          2011
        </span>
      </li>
      <li
        key="2012-2013"
      >
        <span
          className=""
          key="2012"
          onClick={[Function]}
        >
          2012
        </span>
        <span
          className=""
          key="2013"
          onClick={[Function]}
        >
          2013
        </span>
      </li>
      <li
        key="2014-2015"
      >
        <span
          className=""
          key="2014"
          onClick={[Function]}
        >
          2014
        </span>
        <span
          className=""
          key="2015"
          onClick={[Function]}
        >
          2015
        </span>
      </li>
      <li
        key="2016-2017"
      >
        <span
          className=""
          key="2016"
          onClick={[Function]}
        >
          2016
        </span>
        <span
          className=""
          key="2017"
          onClick={[Function]}
        >
          2017
        </span>
      </li>
      <li
        key="2018-2019"
      >
        <span
          className=""
          key="2018"
          onClick={[Function]}
        >
          2018
        </span>
        <span
          className=""
          key="2019"
          onClick={[Function]}
        >
          2019
        </span>
      </li>
      <li
        key="2020-2021"
      >
        <span
          className=""
          key="2020"
          onClick={[Function]}
        >
          2020
        </span>
      </li>
    </ul>
    <div
      className="sidebarTitle"
    >
      Successful Launch
    </div>
    <ul
      className="sidebarList"
    >
      <li>
        <span
          className=""
          onClick={[Function]}
        >
          True
        </span>
        <span
          className=""
          onClick={[Function]}
        >
          False
        </span>
      </li>
    </ul>
    <div
      className="sidebarTitle"
    >
      Successful Landing
    </div>
    <ul
      className="sidebarList"
    >
      <li>
        <span
          className=""
          onClick={[Function]}
        >
          True
        </span>
        <span
          className=""
          onClick={[Function]}
        >
          False
        </span>
      </li>
    </ul>
  </div>
</aside>
`;
